Re-enable the video translation Jasmine tests

andy-armstrong opened the pull request:
https://openedx.atlassian.net/browse/TNL-2689

andy-armstrong commented:
@jzoldak @cahrens Here's the clean up to re-enable the video translation Jasmine tests. It got slightly bigger than I expected because ``AjaxHelpers.server`` didn't quite do what was needed, so I had to refactor it and change the callers, but there were only a handful of them.

andy-armstrong commented:
FYI, I had just a single failure so had to push up a fix, but this next build should be green if the flakey test gods are in a good mood.

cahrens commented:
:+1: Thanks for doing this, Andy.

andy-armstrong commented:
@jzoldak @cahrens The most recent build just got 28 sporadic failures at once. I can't tell if these were caused by my changes, or if there is another such problem going on.

andy-armstrong commented:
I'm going to rerun the tests, so look above for a link to the failing test results.

andy-armstrong commented:
Jenkins test this please

jzoldak commented:
@andy-armstrong that "Could not connect to JSCover server" error I've seen before (tho not often).  I'm pretty sure it is unrelated to your changes.

jzoldak commented on common/static/common/js/spec_helpers/ajax_helpers.js:
```diff
@@ -18,12 +18,13 @@ define(['sinon', 'underscore'], function(sinon, _) {
         });
-        server.respondWith([statusCode, {}, '']);
+        server.respondWith(response);
+        server.autoRespond = true;
```
The sinon docs [say](http://sinonjs.org/docs/#responses):

> Boolean request.autoRespond
> When set to true, causes the server to automatically respond to incoming requests after a timeout. The default timeout is 10ms but you can control it through the autoRespondAfter property. Note that this feature is intended to help during mockup development, and is not suitable for use in tests.

If we need to set autoRespond to true, can you please add a comment as to why?

andy-armstrong commented:
@jzoldak I have a green build on a re-run. Good to merge?

jzoldak commented:
@andy-armstrong Why are you setting server.autoRespond to true? That will add 10ms to every server response.

andy-armstrong commented:
@jzoldak Thanks for catching that. This setting was originally done in the videolist_spec, and I moved it into the AjaxHelpers method because I misunderstood what it was doing (I thought it would auto respond to the request, and that seemed like a useful default). 

I've moved the setting back so that it is only done by the video tests (I tried running without it but they fail, so I think they depend upon the latency somehow). I've squashed commits to save time, but if you look at ajax_helpers.js now you'll see the change is gone.

jzoldak commented:
:+1:

andy-armstrong commented:
Jenkins, test this please
